torch/csrc/distributed/c10d/NCCLUtils.cpp

```diff
@@ -26,9 +26,36 @@ ncclComm_t NCCLComm::getNcclComm() {
             ". ",
             commFailureMsg));
   }
+  // only wait for initialization if nonblocking mode is enabled
+  if (nccl_use_nonblocking()) {
+    waitUtilInitialized(nccl_nonblocking_timeout());
+  }
+
   return ncclComm_;
 }
 
+void NCCLComm::waitUtilInitialized(int timeoutSecs) {
```

Comment:
this seems pretty similar to `ProcessGroupNCCL::WorkNCCL::synchronizeInternal` but I don't think that code would be easy to reuse or refactor.  But there are some things perhaps to borrow from it (like sleeping rather than busy looping).